### ISSUE TITLE
Cifar10 accuracy with resnet18 on a single client is stuck at 75%

### ISSUE THREAD

**Issue by shahdev** [OpenMined/PySyft#3786]
## Description
Cifar10 accuracy with resnet18 on a single client is stuck at 75%
I ran for 200 epochs with a learning rate schedule of (0.1 till epoch100, 0.01 from 100-150, 0.001 from 150-200)
The accuracy I achieve on the test set is around 75%
Cifar10 accuracy with resnet18 should reach around 88-90% on centralized data.

I used the same hyper-parameters as in centralized implementation

## How to Reproduce
here is my code for a single client execution

import syft as sy  # <-- NEW: import the Pysyft library
import torch
torch.set_default_tensor_type(torch.cuda.FloatTensor)
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
from torchvision import datasets, transforms, models

hook = sy.TorchHook(torch)  # <-- NEW: hook PyTorch ie add extra functionalities to support Federated Learning
bob = sy.VirtualWorker(hook, id="bob")  # <-- NEW: define remote worker bob

### modifying lr during training
def lr_function(epoch, first_lr_drop, second_lr_drop, third_lr_drop, initial_lr):
	if (epoch >= third_lr_drop):
		return initial_lr*0.1*0.1*0.1
	elif (epoch >= second_lr_drop):
		return initial_lr*0.1*0.1
	elif (epoch >= first_lr_drop):
		return initial_lr*0.1
	else:
		return initial_lr

class Arguments():
	def __init__(self):
		self.batch_size = 64
		self.test_batch_size = 1000
		self.epochs = 200
		self.lr = 0.1
		self.momentum = 0.9
		self.no_cuda = False
		self.seed = 1
		self.log_interval = 100
		self.save_model = False
		self.first_lr_drop = 100
		self.second_lr_drop = 150
		self.third_lr_drop = 200

args = Arguments()

use_cuda = not args.no_cuda and torch.cuda.is_available()

torch.manual_seed(args.seed)

device = torch.device("cuda" if use_cuda else "cpu")

transform = transforms.Compose(
	[ transforms.RandomCrop(32, padding=4),
	transforms.RandomHorizontalFlip(),
	transforms.ToTensor(),
	#transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
	])

transform_test = transforms.Compose([
		transforms.ToTensor()
	])

federated_train_loader = sy.FederatedDataLoader( # <-- this is now a FederatedDataLoader 
	datasets.CIFAR10('../data', train=True, download=True,
				   transform=transform)
	.federate([bob]), # <-- NEW: we distribute the dataset across all the workers, it's now a FederatedDataset
	batch_size=args.batch_size, shuffle=True)

test_loader = torch.utils.data.DataLoader(
	datasets.CIFAR10('../data', train=False, transform=transform_test),
	batch_size=args.test_batch_size, shuffle=True)

print(device)
model = models.resnet18(pretrained=False)
model.fc = nn.Linear(in_features=model.fc.in_features, out_features=10, bias=True)
model = model.to(device)

bobs_model = model.copy().send(bob) 

bobs_criterion = nn.CrossEntropyLoss()

criterion = nn.CrossEntropyLoss()

for epoch in range(args.epochs):
	bobs_model.train()

	bobs_opt = optim.SGD(params=bobs_model.parameters(),lr=args.lr, momentum=args.momentum, weight_decay=2e-4)

	for param_group in bobs_opt.param_groups:
		param_group['lr'] = lr_function(epoch, args.first_lr_drop, args.second_lr_drop, args.third_lr_drop, args.lr)

	for batch_idx, (data, target) in enumerate(federated_train_loader): # <-- now it is a distributed dataset
		location = data.location
		data, target = data.to(device), target.to(device)
		bobs_opt.zero_grad()
		bobs_output = bobs_model(data)
		bobs_loss = bobs_criterion(bobs_output, target)
		bobs_loss.backward()
		bobs_opt.step()
		loss = bobs_loss.detach().get()

		if batch_idx % args.log_interval == 0:
			print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
				epoch, batch_idx * args.batch_size, len(federated_train_loader) * args.batch_size,
				100. * batch_idx / len(federated_train_loader), loss.item()), flush=True)

	bobs_model.get()

	test_loss = 0
	correct = 0
	with torch.no_grad():
		for data, target in test_loader:
			data, target = data.to(device), target.to(device)
			output = bobs_model(data)
			test_loss += criterion(output, target).item() # sum up batch loss
			pred = output.argmax(1, keepdim=True) # get the index of the max log-probability 
			correct += pred.eq(target.view_as(pred)).sum().item()

	test_loss /= len(test_loader.dataset)

	print('\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n'.format(
		test_loss, correct, len(test_loader.dataset),
		100. * correct / len(test_loader.dataset)))

	bobs_model.send(bob)

## Expected Behavior
I run the code with a single client 
Output after 200 epochs:

Optimizer lr:  0.0010000000000000002
Train batch size:  128
Train Epoch: 199 [0/50048 (0%)]	Loss: 0.000341
Train Epoch: 199 [6400/50048 (13%)]	Loss: 0.000293
Train Epoch: 199 [12800/50048 (26%)]	Loss: 0.000387
Train Epoch: 199 [19200/50048 (38%)]	Loss: 0.000356
Train Epoch: 199 [25600/50048 (51%)]	Loss: 0.000346
Train Epoch: 199 [32000/50048 (64%)]	Loss: 0.000281
Train Epoch: 199 [38400/50048 (77%)]	Loss: 0.000394
Train Epoch: 199 [44800/50048 (90%)]	Loss: 0.000462
Training avg batch loss: 0.0004
Epoch time:  196.71481251716614
Number of batches in epoch:  390

Test set: avg cln loss: 1.2113, cln acc: 7599/10000 (76%)

## System Information
 - Linux (Ubuntu)
 - Red Hat Enterprise Linux Server release 7.6
 - Python 3.7
 - Conda 4.8.3
